### PR TITLE
feat(build): preserve repository name case for build upload

### PR DESCRIPTION
**Pull request by runningcode**
## Summary
- Preserve original case of repository names in build upload command
- Add case-preserving VCS functions that use simple regex extraction
- Only affects build upload command, other commands unchanged

## Changes
- Added `get_repo_from_remote_preserve_case()` function with regex-based case preservation
- Added `git_repo_base_repo_name_preserve_case()` wrapper function  
- Updated build upload command to use case-preserving functions for both head and base repo names
- Added comprehensive test coverage for various Git URL patterns

## Technical Approach
Instead of duplicating complex VCS parsing logic, this uses a hybrid approach:
1. Uses existing parsing for correctness and edge case handling
2. Applies simple regex to extract case-preserved repository path
3. Falls back to lowercase version if extraction fails

This keeps the implementation simple while maintaining robustness.

## Test Coverage
- GitHub, GitLab, Bitbucket URLs (HTTPS and SSH)
- Edge cases and fallback behavior
- Verification that other commands still use lowercase

Fixes EME-312

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/utils/vcs.rs [hunk 1]
```diff
@@ -196,13 +220,21 @@ impl VcsUrl {
         if let Some(caps) = BITBUCKET_SERVER_PATH_RE.captures(path) {
             return VcsUrl {
                 provider: host.to_lowercase(),
-                id: format!("{}/{}", &caps[1].to_lowercase(), &caps[2].to_lowercase()),
+                id: if preserve_case {
```
[the remark this answers is not in the document]
Comment (runningcode) [reply src/utils/vcs.rs]: I don't see how this is related to GCB.

File: src/utils/vcs.rs [hunk 2]
```diff
@@ -129,6 +133,13 @@ impl VcsUrl {
         }
     }
 
+    fn into_lowercase(self) -> VcsUrl {
```
Comment (runningcode) [inline src/utils/vcs.rs]: I had these two clippy warning with function you suggested so I adjusted it to what I have in the code.


``` 
warning: methods with the following characteristics: (`to_*` and `self` type is not `Copy`) usually take `self` by reference
        --> src/utils/vcs.rs:136:21
         |
     136 |     fn to_lowercase(mut self) -> VcsUrl {
         |                     ^^^^^^^^
         |
         = help: consider choosing a less ambiguous name
         = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
         = note: `#[warn(clippy::wrong_self_convention)]` on by default
```

```
  warning: methods with the following characteristics: (`to_*` and `self` type is not `Copy`) usually take `self` by reference
        --> src/utils/vcs.rs:136:21
         |
     136 |     fn to_lowercase(self) -> VcsUrl {
         |                     ^^^^
         |
         = help: consider choosing a less ambiguous name
         = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
         = note: `#[warn(clippy::wrong_self_convention)]` on by default
```

**Comment by runningcode** [comment]
Thanks @szokeasaurusrex that’s a good suggestion. I think I applied it as you suggested with one exception noted in my comment above. Let me know if there is a better alternative.

File: src/utils/vcs.rs (hunk 2 above)
Comment (szokeasaurusrex) [reply src/utils/vcs.rs]: Yeah, if clippy complained, then this is the right thing to do!

File: src/utils/vcs.rs [hunk 3]
```diff
@@ -129,6 +133,13 @@ impl VcsUrl {
...
+        VcsUrl {
+            provider: self.provider,
+            id: self.id.to_lowercase(),
+        }
+    }
```
Comment (szokeasaurusrex) [inline src/utils/vcs.rs]: I think Clippy would also accept the following, which is a bit cleaner imo. But, I have not tested it. What you have already is also completely OK 🙂 

```suggestion
    fn into_lowercase(mut self) -> VcsUrl {
        self.id = self.id.to_lowercase();
        self
    }
```

Comment (szokeasaurusrex) [reply src/utils/vcs.rs]: From what I can tell, Clippy's only complaint was that the function was called `to_...` and not `into_...`

Comment (runningcode) [reply src/utils/vcs.rs]: Yup, fixed!
